src/components/core/Button/Button.module.scss

```diff
@@ -8,10 +8,38 @@
   color: $color-white;
 
   font-size: $fs-base;
-  border: 2px solid $color-primary;
   border-radius: $space-m;
   cursor: pointer;
 
+  &--outline {
+    &--solid {
+      border: 2px solid $color-primary;
+    }
+
+    &--none {
+      border: none;
+    }
+  }
```

Comment:
This can't be done in this way, there is no nested modifiers in BEM. All modifiers go on end